modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/views/table/dnd_table/ContextProvider.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {useCallback, useMemo, useState} from 'react';
+
+import Context from './TableContext';
+
+function ContextProvider({children, columnsNames}) {
+	const [tableWidth, updateTableWidth] = useState(null);
+	const [columnsDefinitions, updateColumnsDefinitions] = useState(new Map());
+	const [draggingColumnName, updateDraggingColumnName] = useState(null);
+	const [draggingAllowed, updateDraggingAllowed] = useState(true);
+
+	const registerColumn = (name, width, resizable) => {
+		updateColumnsDefinitions((definitions) => {
+			const updatedDefinitions = new Map(definitions);
+
+			updatedDefinitions.set(name, {
+				resizable,
+				width,
+			});
+
+			return updatedDefinitions;
+		});
+	};
+
+	const isFixed = useMemo(() => {
+		const columnsAreAllRegistered = columnsNames.reduce(
+			(registered, name) => {
+				return registered && !!columnsDefinitions.get(name);
+			},
+			true
+		);
+
+		return columnsAreAllRegistered;
+	}, [columnsDefinitions, columnsNames]);
+
+	const resizeColumn = useCallback(
+		(name, width) => {
+			if (isFixed) {
+				updateColumnsDefinitions((definitions) => {
+					const resizedColumn = definitions.get(name);
+
+					const isColumnReducing = resizedColumn.width > width;
+
+					let totalColsWidth = 0;
+
+					definitions.forEach((definition) => {
+						totalColsWidth += definition.width;
+					});
```

Comment:
Seeing as you are using `Map` here, it's possible that you'll incur slightly less runtime overhead by switching to `for`/`of` (ie. avoiding the repeated function calls):

```js
for (const {width} of definitions.values()) {
    totalWidth += width;
} 
```

Reply:
I'll bear it in mind! At the moment I'll keep it as it is due to the `no-for-of-loops` rule which is currently active on portal

Reply:
I thought we turned it off on `master` already (it was only in there for IE), but we probably just created a ticket for it... (which also, I can't find).

Reply:
Created quite-possibly-duplicate-but-we-will-see issue: https://github.com/liferay/liferay-frontend-projects/issues/405